File: frontend/src/domain/types/controlUnit.ts
import type { ControlResource } from './controlResource'
import type { Undefine } from '@mtes-mct/monitor-ui'
import type { Except } from 'type-fest'

export namespace ControlUnit {
  export interface ControlUnit {
    administration: string
    contact: string | undefined
    id: number
    name: string
    resources: ControlResource[]
  }

  export type ControlUnitData = Except<ControlUnit, 'id'>

  export type ControlUnitDraft = Undefine<ControlUnitData>
}

File: backend/src/main/kotlin/fr/gouv/cnsp/monitorfish/domain/entities/mission/ControlUnit.kt
package fr.gouv.cnsp.monitorfish.domain.entities.mission

import kotlinx.serialization.Serializable

@Serializable
data class ControlUnit(
    val id: Int,
    val administration: String,
    val isArchived: Boolean,
    val name: String,
    val resources: List<ControlResource>,
    val contact: String? = null,
)
